src/app/components/canvas/canvas.component.ts

```diff
@@ -317,9 +324,69 @@ export class CanvasComponent implements OnChanges, OnInit, OnDestroy {
     }
 
     persistLayerPosition(elements: Array<HTMLElement>): void {
-        this._draggingService.setDrawingArea(this.drawingArea);
-        this._displayService.setCoordsInfo(
-            this._draggingService.getCoordinates(elements)
+        const coordinates = this.getCoordinates(elements);
+        this.coordinateChanged.next(coordinates);
+
+        if (this.persistUiChangesForRunInTextarea) {
```

Comment:
Das Setzen in der Textarea sollte nur stattfinden, wenn wir nicht in der Merge-View sind.